pkg/binding/binding.go

```diff
@@ -64,15 +66,11 @@ func (o *BindingClient) GetFlags(flags map[string]string) map[string]string {
 }
 
 func (o *BindingClient) GetServiceInstances(namespace string) (map[string]unstructured.Unstructured, error) {
-	isServiceBindingInstalled, err := o.kubernetesClient.IsServiceBindingSupported()
+	err := o.checkServiceBindingOperatorInstalled()
```

Comment:
Putting this over here means this method is also doing some validation that's not about "getting service instances". How about checking this in the `Validate` method at CLI layer?

Reply:
I had also added the check in the `ValidateAddBinding` method (which is indeed called by the `Validate` method in the CLI layer). See `pkg/binding/add.go` in the last commit ([64bd15fc733d04b1592e0aab5a66250d864f681c](https://github.com/redhat-developer/odo/pull/5936/commits/64bd15fc733d04b1592e0aab5a66250d864f681c))

I chose to leave the check here in `GetServiceInstances` (I think this was already there in the original implementation of `odo add binding`), because I think it still makes sense as it relies on the ServiceBinding operator presence to get this list. I know we might not be there yet, but it also made sense to me especially if this business package could potentially be imported and used by external tools.

Does that make sense?

Reply:
> because I think it still makes sense as it relies on the ServiceBinding operator presence to get this list. 

Ah, I forgot that we are trying to get the bindable service instances. OK, now it makes sense to put this in here. 👍🏾 

> I know we might not be there yet, but it also made sense to me especially if this business package could potentially be imported and used by external tools.

That's the (indirect) goal. Some day, for sure. :)

/lgtm